pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -247,6 +248,12 @@ protected TopicStatsHelper initialValue() {
 
     // Record the last time a data message (ie: not an internal Pulsar marker) is published on the topic
     private volatile long lastDataMessagePublishedTimestamp = 0;
+
+    // Record the total expired subscription, if enable subscription expiration feature
+    private static final AtomicLongFieldUpdater<PersistentTopic> EXPIRED_SUBSCRIPTION_NUMBERS_UPDATER =
+            AtomicLongFieldUpdater.newUpdater(PersistentTopic.class, "expiredSubscriptionNumbers");
+    private volatile long expiredSubscriptionNumbers = 0L;
```

Comment:
Seems it's no need to add this

Reply:
agree, it is added only for unit test. now i figure out another way to check the logic. PTAL again